lucetc/tests/wasm.rs

```diff
@@ -79,18 +71,22 @@ mod module_data {
         .unwrap();
         let h = HeapSettings::default();
         let c = Compiler::new(&m, OptLevel::Best, &b, h).expect("compile icall");
-        let _module_data = c.module_data().unwrap();
+        let mdata = c.module_data().unwrap();
 
-        /*  TODO can't express these with module data
-        assert_eq!(p.import_functions().len(), 1);
-        assert_eq!(p.import_functions()[0].module(), "env");
-        assert_eq!(p.import_functions()[0].field(), "icalltarget");
-        assert_eq!(p.globals().len(), 0);
-        assert_eq!(p.defined_functions().len(), 4);
+        assert_eq!(mdata.import_functions().len(), 1);
+        assert_eq!(mdata.import_functions()[0].module, "env");
+        assert_eq!(
+            mdata.sym_for(mdata.import_functions()[0].fn_idx),
+            Some(&b"icalltarget"[..])
+        );
+        assert_eq!(mdata.function_info().len(), 5);
         assert_eq!(
-            p.defined_functions().get(0).unwrap().symbol(),
```

Comment:
This goes from `0` to `1` because `function_info` also contains entries for imported functions. I imagine dynamic linking-style function import would look like walking imports, looking for those indices, and updating the corresponding `FunctionSpec` with a pointer that would otherwise have had a relocation anyway.